mslib/msui/mscolab.py

```diff
@@ -276,15 +277,21 @@ def login_handler(self):
             self.httpBb.accepted.connect(self.login_server_auth)
             self.httpBb.rejected.connect(lambda: self.stackedWidget.setCurrentWidget(self.loginPage))
         else:
-            save_password_to_keyring(service_name="MSCOLAB", username=emailid, password=password)
+            try:
+                save_password_to_keyring(service_name="MSCOLAB", username=emailid, password=password)
+            except (NoKeyringError, PasswordSetError, InitError) as ex:
+                logging.warning("Can't use Keyring on your system: %s" % ex)
```

Comment:
It can be a "user" error, for example I have setup to use my gpg key. When I not have added the passphrase and clicked it away I have a keyring but can't access it. This give the PasswordSetError. The NoKeyringError happens because in our docker container we don't have gnome-keyring and when the container is started without --cap-add=IPC_LOCK the dbus session did not work correctly I got the InitError.

Warning  get printedon the terminal when it is started from there. By docker hopefully the description helps to ommit these mistakes. All these problems are related to the users system or desktop configuration.